src/createReducer.test.ts

```diff
@@ -172,6 +274,121 @@ describe('createReducer', () => {
       )
     })
   })
+
+  describe('builder "addMatcher" method', () => {
+    const prepareNumberAction = (payload: number) => ({
+      payload,
+      meta: { type: 'number_action' }
+    })
+    const prepareStringAction = (payload: string) => ({
+      payload,
+      meta: { type: 'string_action' }
+    })
+
+    const numberActionMatcher = (a: AnyAction): a is PayloadAction<number> =>
+      a.meta.type === 'number_action'
+    const stringActionMatcher = (a: AnyAction): a is PayloadAction<string> =>
+      a.meta.type === 'string_action'
+
+    const incrementBy = createAction('increment', prepareNumberAction)
+    const decrementBy = createAction('decrement', prepareNumberAction)
+    const concatWith = createAction('concat', prepareStringAction)
+
+    const initialState = { numberActions: 0, stringActions: 0 }
+
+    test('uses the reducer of matching actionMatchers', () => {
+      const reducer = createReducer(initialState, builder =>
+        builder
+          .addMatcher(numberActionMatcher, state => {
+            state.numberActions += 1
+          })
+          .addMatcher(stringActionMatcher, state => {
+            state.stringActions += 1
+          })
+      )
+      expect(reducer(undefined, incrementBy(1))).toEqual({
+        numberActions: 1,
+        stringActions: 0
+      })
+      expect(reducer(undefined, decrementBy(1))).toEqual({
+        numberActions: 1,
+        stringActions: 0
+      })
+      expect(reducer(undefined, concatWith('foo'))).toEqual({
+        numberActions: 0,
+        stringActions: 1
+      })
+    })
+    test('prefers explicit reducer cases over actionMatchers', () => {
```

Comment:
Question on the possible semantics: what if we want to have _both_ the exact match _and_ the wildcard match run?  Is that a desirable behavior?

Reply:
We could even go so far as to run all matchers, but all that would be pretty un-reducer-like behaviour for my intuition.

But generally, I'm open for discussion there.

The question would be in which order they run.

Reply:
https://github.com/redux-utilities/reduce-reducers is totally a thing, and in fact I'd even talked about adding it to RTK at some point.  Presumably they'd run in the order they were defined.